google/resource_storage_bucket_object.go

```diff
@@ -92,6 +95,7 @@ func resourceStorageBucketObject() *schema.Resource {
 				Optional:      true,
 				ForceNew:      true,
 				ConflictsWith: []string{"content"},
+				StateFunc:     createSourceState,
```

Comment:
How about writing an anonymous function here that turns the `interface{}` into a string, and then renaming `createSourceState` to something like `filenameWithMd5` with the signature:
```go
func filenameWithMd5(filename string) string {
   // ...
}
```

That way the function is a bit more general and better named.

Reply:
Good suggestion. Changed.